core/src/main/scala/org/apache/spark/rdd/CoGroupedRDD.scala

```diff
@@ -58,10 +58,20 @@ private[spark] case class NarrowCoGroupSplitDep(
  *                   narrowDeps should always be equal to the number of parents.
  */
 private[spark] class CoGroupPartition(
-    idx: Int, val narrowDeps: Array[Option[NarrowCoGroupSplitDep]])
+    override val index: Int, val narrowDeps: Array[Option[NarrowCoGroupSplitDep]])
```

Comment:
This may upset MiMa since you effectively remove an 'idx' property but it can be ignored for our purposes as it's a private class.